runtime/src/main/java/org/corfudb/runtime/view/SMRObject.java

```diff
@@ -0,0 +1,150 @@
+package org.corfudb.runtime.view;
+
+import com.google.common.reflect.TypeToken;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.object.CorfuCompileProxy;
+import org.corfudb.runtime.object.CorfuCompileWrapperBuilder;
+import org.corfudb.runtime.object.ICorfuSMR;
+import org.corfudb.util.serializer.ISerializer;
+import org.corfudb.util.serializer.Serializers;
+
+import java.util.UUID;
+
+/**
+ * Created by mwei on 4/6/16.
+ */
+@Slf4j
+@Getter
+@Builder(builderClassName = "Builder")
```

Comment:
Its just syntactic sugar. By default it will be named SMRObject.SMRObjectBuilder which I find a little superfluous, so instead this will generate a SMRObject.Builder instead.